### PR TITLE
AWP-1628: Add DropdownButton component with types and functionalities

### PR DESCRIPTION
**Pull request by global-usman**
[AWP-1628](https://team-1637335847501.atlassian.net/jira/software/projects/AWP/issues/AWP-1628)

- Implemented DropdownButton component with select and button types
- Added styles for DropdownButton component
- Included storybook stories and unit tests for DropdownButton component
- Updated SvgIcon component to include ChevronDown icon

[AWP-1628]: https://team-1637335847501.atlassian.net/browse/AWP-1628?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/dropdownButton/dropdownButton.tsx
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx:1]: Can we rename this to `DropdownButton.tsx`?

File: src/components/dropdownButton/dropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+const DropdownButton = <T,>({
+  disabled = false,
+  type,
+  options,
+  onSelect,
+  renderOption,
+  label,
+}: DropdownButtonProps<T>) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
```
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx]:
```suggestion
  const [isOpen, setIsOpen] = useState(false);
```

File: src/components/dropdownButton/dropdownButton.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  const [selectedOption, setSelectedOption] = useState<T | null>(
+    options.at(0) || null,
+  );
+  const dropdownRef = useRef<HTMLDivElement>(null);
+  const [position, setPosition] = useState<"top" | "bottom">("bottom");
```
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx]: can we make a type for top | bottom?

File: src/components/dropdownButton/dropdownButton.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+  useEffect(() => {
+    if (dropdownRef.current) {
```
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx]:
```suggestion
    if (dropdownRef.current && isOpen) {
```

File: src/components/dropdownButton/dropdownButton.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            fill={
+              disabled
+                ? "var(--neutral-200, #9CA3AF)"
+                : "var(--primary-50, #2AACE2)"
+            }
+          />
+        </IconWrapper>
+      </CustomSelectButton>
+      {isOpen && (
+        <SelectItemsWrapper position={position === "top" ? 45 : undefined}>
```
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx]: Can we this position condition in the styled component for better readability?

File: src/components/dropdownButton/dropdownButton.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+  onSelect?: (option: T) => void;
+  label?: string;
+}
+
+const DropdownButton = <T,>({
+  disabled = false,
+  type,
+  options,
+  onSelect,
+  renderOption,
```
Comment (zamanaf) [inline src/components/dropdownButton/dropdownButton.tsx]: Should we have a default render option method?

File: src/components/dropdownButton/__docs__/DropdownButton.mdx [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+#### Arguments
+
+- **type** _`"select" | "button"`_ -  Type of the dropdown button.
+- **options** _`T[]`_ - An array of options to be displayed in the dropdown.
+- **onSelect** _`(option: T) => void`_ - A function that is called whenever an option is selected. It receives the selected option as a parameter.
+- **renderOption** _`(option: T | null) => ReactNode`_ - A function that returns the JSX to render for each option.
+- **disabled** _`boolean`_ - Whether the dropdown button is disabled or not.
+- **label** _`string`_ - Label text for the dropdown button.
+- **...rest** - Additional HTML attributes that can be passed to the underlying button element.
```
Comment (zamanaf) [inline src/components/dropdownButton/__docs__/DropdownButton.mdx]:
```suggestion
```

File: src/components/dropdownButton/DropdownStyledComponents.tsx
Comment (darnold001) [inline src/components/dropdownButton/DropdownStyledComponents.tsx:1]: Can we rename this to `dropdown` as this is more than just a button?

File: src/components/dropdownButton/DropdownStyledComponents.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
+import styled from "styled-components";
+import { DropdownPosition } from "./DropdownButton";
+
+export const SelectWrapper = styled.div`
+  display: flex;
+  padding: 8px 16px;
```
Comment (darnold001) [inline src/components/dropdownButton/DropdownStyledComponents.tsx]: Let's try and use REMs when possible (.5rem , 1rem)

**Review by darnold001 — APPROVED** [review]
Looks good, just had a few comments about naming and rem/pixels

**Pull request merged**